static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -0,0 +1,213 @@
+    offering: string,
+    campaignId: string,
+    variantId: string
+): string => {
+    let prefix = '';
+    const offerings = membershipEngagementBannerUtils.offerings;
+
+    // mem and cont chosen to be consistent with default campaign code prefixes.
+    if (offering === offerings.membership) {
+        prefix = 'mem';
+    } else if (offering === offerings.contributions) {
+        prefix = 'cont';
+    }
+
+    return `${prefix}_${campaignId}_${variantId}`;
+};
+
+const getUserVariantParams = (
+    userVariant: ?Variant,
+    campaignId: ?string,
+    defaultOffering: string
+): Object => {
+    if (campaignId && userVariant && userVariant.engagementBannerParams) {
+        const userVariantParams: any = userVariant.engagementBannerParams;
+
+        if (!userVariantParams.campaignCode) {
+            const offering: string =
+                userVariantParams.offering || defaultOffering;
+
+            userVariantParams.campaignCode = buildCampaignCode(
+                offering,
+                campaignId,
+                userVariant.id
+            );
+        }
+
+        return userVariantParams;
+    }
+    return {};
+};
+
+/*
+ * Params for the banner are overlaid in this order, earliest taking precedence:
+ *
+ *  * Variant (if the user is in an A/B testing variant)
+ *  * Edition
+ *  * Offering ('membership' or 'contributions')
+ *  * Default
+ *
+ * The 'offering' in use comes from either:
+ *
+ *  * Variant (if the user is in an A/B testing variant)
+ *  * Edition (only one offering can be the default for a given Edition)
+ *
+ * Returns either 'null' if no banner is available for this edition,
+ * otherwise a populated params object that looks like this:
+ *
+ *  {
+ *    minArticles: 5, // how many articles should the user see before they get the engagement banner?
+ *    messageText: "..."
+ *    colourStrategy: // a function to determine what css class to use for the banner's colour
+ *    buttonCaption: "Become a Supporter"
+ *  }
+ *
+ */
+const deriveBannerParams = (location: string): ?Object => {
+    const defaultParams = membershipEngagementBannerUtils.defaultParams(
+        location
+    );
+    const userTest = getUserTest();
+    const campaignId = userTest ? userTest.campaignId : undefined;
+    const userVariant = getUserVariant(userTest);
+
+    if (userVariant && userVariant.blockEngagementBanner) {
+        return;
+    }
+
+    return Object.assign(
+        {},
+        defaultParams,
+        getUserVariantParams(userVariant, campaignId, defaultParams.offering)
+    );
+};
+
+// Used to send an interaction if the engagement banner is shown.
+// #? A lot of null checking here, but it's hard to reason about the possible values
+//    `interaction` could take. Perhaps we can simplify when we have more Flow coverage
+const recordInteraction = (interaction: ?Interaction): void => {
+    if (interaction) {
+        const { component, value } = interaction;
+
+        if (component && value) {
```

Comment:
I was being overly cautious by not removing these checks, as it is difficult to reason about this code. However, I think you're right, this check is unnecessary